### ISSUE TITLE
RHEL / Centos 7 libgcrypt does not support GCRY_CIPHER_MODE_GCM

### ISSUE THREAD

**Issue by atomicturtle** [greenbone/gvm-libs#107]
This is probably better handled as a version check in cmake for a compatible libgcrypt version. Outside of including external libraries in the archive (we do this in the ossec project) I'm not sure what you can do to fix this:

[root@c7-64-dev-scott gvm-libs-9.0.2]# rpm -q libgcrypt
libgcrypt-1.5.3-14.el7.x86_64

[ 42%] Building C object nasl/CMakeFiles/openvas_nasl_shared.dir/nasl_crypto2.c.o
cd /root/src/gvm-libs-9.0.2/nasl && /usr/lib64/ccache/cc -DHAVE_LIBKSBA -Dopenvas_nasl_shared_EXPORTS -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include  -Wformat -Wformat-security -O2 -D_FORTIFY_SOURCE=2 -fstack-protector -Wall -D_FILE_OFFSET_BITS=64 -DLARGEFILE_SOURCE=1 -fno-strict-aliasing -g -Werror -fPIC   -DOPENVASSD_CONF=\"/etc/openvas/openvassd.conf\" -DOPENVAS_STATE_DIR=\"/var/lib/openvas\" -DOPENVASLIB_VERSION=\"9.0.2\" -DOPENVAS_SYSCONF_DIR=\"/etc/openvas\" -o CMakeFiles/openvas_nasl_shared.dir/nasl_crypto2.c.o   -c /root/src/gvm-libs-9.0.2/nasl/nasl_crypto2.c
/root/src/gvm-libs-9.0.2/nasl/nasl_crypto2.c: In function ‘nasl_aes128_gcm_encrypt’:
/root/src/gvm-libs-9.0.2/nasl/nasl_crypto2.c:1572:51: error: ‘GCRY_CIPHER_MODE_GCM’ undeclared (first use in this function)
   return encrypt_data (lexic, GCRY_CIPHER_AES128, GCRY_CIPHER_MODE_GCM);
                                                   ^
/root/src/gvm-libs-9.0.2/nasl/nasl_crypto2.c:1572:51: note: each undeclared identifier is reported only once for each function it appears in
/root/src/gvm-libs-9.0.2/nasl/nasl_crypto2.c: In function ‘nasl_aes256_gcm_encrypt’:
/root/src/gvm-libs-9.0.2/nasl/nasl_crypto2.c:1578:51: error: ‘GCRY_CIPHER_MODE_GCM’ undeclared (first use in this function)
   return encrypt_data (lexic, GCRY_CIPHER_AES256, GCRY_CIPHER_MODE_GCM);
                                                   ^
/root/src/gvm-libs-9.0.2/nasl/nasl_crypto2.c: In function ‘nasl_aes128_gcm_encrypt’:
/root/src/gvm-libs-9.0.2/nasl/nasl_crypto2.c:1573:1: error: control reaches end of non-void function [-Werror=return-type]
 }
 ^
/root/src/gvm-libs-9.0.2/nasl/nasl_crypto2.c: In function ‘nasl_aes256_gcm_encrypt’:
/root/src/gvm-libs-9.0.2/nasl/nasl_crypto2.c:1579:1: error: control reaches end of non-void function [-Werror=return-type]
 }
 ^
cc1: all warnings being treated as errors
make[2]: *** [nasl/CMakeFiles/openvas_nasl_shared.dir/nasl_crypto2.c.o] Error 1
make[2]: Leaving directory `/root/src/gvm-libs-9.0.2'
make[1]: *** [nasl/CMakeFiles/openvas_nasl_shared.dir/all] Error 2
make[1]: Leaving directory `/root/src/gvm-libs-9.0.2'
make: *** [all] Error 2

**Comment by cfi-gb** [comment]
Looks like a duplicate of https://github.com/greenbone/gvm-libs/issues/40 where the cmake check is already included in the upcoming releases after 9.0.2:

https://github.com/greenbone/gvm-libs/pull/41

https://github.com/greenbone/openvas-scanner/pull/35
